### ISSUE TITLE
Node v12: CLI Tool  Remote tool installation fails

### ISSUE THREAD

**Issue by shirshak55** [Chocobozzz/PeerTube#1877]
As per docs i went to https://docs.joinpeertube.org/#/maintain-tools?id=remote-tools and I have fresh ubuntu will all node etc installed. And ran these commands
```
$ git clone https://github.com/Chocobozzz/PeerTube.git
$ CLONE="$(pwd)/PeerTube"
$ cd ${CLONE}
```

After that I ran 
```
$ NOCLIENT=1 yarn install --pure-lockfile
```

Now it is giving this error
```
error /home/zyzzzz/PeerTube/node_modules/libxmljs: Command failed.
Exit code: 1
Command: node-pre-gyp install --fallback-to-build --loglevel http
Arguments:
Directory: /home/zyzzzz/PeerTube/node_modules/libxmljs
Output:
node-pre-gyp WARN Using request for node-pre-gyp https download
node-pre-gyp http GET https://github.com/libxmljs/libxmljs/releases/download/v0.19.5/node-v72-linux-x64.tar.gz
node-pre-gyp http 404 https://github.com/libxmljs/libxmljs/releases/download/v0.19.5/node-v72-linux-x64.tar.gz
node-pre-gyp WARN Tried to download(404): https://github.com/libxmljs/libxmljs/releases/download/v0.19.5/node-v72-linux-x64.tar.gz
node-pre-gyp WARN Pre-built binaries not found for libxmljs@0.19.5 and node@12.4.0 (node-v72 ABI, glibc) (falling back to source compile with node-gyp)
node-pre-gyp http 404 status code downloading tarball https://github.com/libxmljs/libxmljs/releases/download/v0.19.5/node-v72-linux-x64.tar.gz
gyp info it worked if it ends with ok
gyp info using node-gyp@3.8.0
gyp info using node@12.4.0 | linux | x64
gyp info ok
make: Entering directory '/home/zyzzzz/PeerTube/node_modules/libxmljs/build'
  CXX(target) Release/obj.target/xmljs/src/libxmljs.o
In file included from ../node_modules/nan/nan_converters.h:67:0,
                 from ../node_modules/nan/nan.h:202,
                 from ../src/libxmljs.h:7,
                 from ../src/libxmljs.cc:7:
../node_modules/nan/nan_converters_43_inl.h: In static member function ‘static Nan::imp::ToFactoryBase<v8::Boolean>::return_t Nan::imp::ToFactory<v8::Boolean>::convert(v8::Local<v8::Value>)’:
../node_modules/nan/nan_converters_43_inl.h:18:51: warning: ‘v8::MaybeLocal<v8::Boolean> v8::Value::ToBoolean(v8::Local<v8::Context>) const’ is deprecated: ToBoolean can never throw. Use Local version. [-Wdeprecated-declarations]
       val->To ## TYPE(isolate->GetCurrentContext())                            \
                                                   ^
../node_modules/nan/nan_converters_43_inl.h:22:1: note: in expansion of macro ‘X’
 X(Boolean)
 ^
In file included from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8-internal.h:14:0,
                 from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:25,
                 from ../src/libxmljs.cc:3:
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:2524:63: note: declared here
                     V8_WARN_UNUSED_RESULT MaybeLocal<Boolean> ToBoolean(
                                                               ^
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8config.h:322:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../node_modules/nan/nan_converters.h:67:0,
                 from ../node_modules/nan/nan.h:202,
                 from ../src/libxmljs.h:7,
                 from ../src/libxmljs.cc:7:
../node_modules/nan/nan_converters_43_inl.h: In static member function ‘static Nan::imp::ValueFactoryBase<bool>::return_t Nan::imp::ToFactory<bool>::convert(v8::Local<v8::Value>)’:
../node_modules/nan/nan_converters_43_inl.h:37:57: warning: ‘v8::Maybe<bool> v8::Value::BooleanValue(v8::Local<v8::Context>) const’ is deprecated: BooleanValue can never throw. Use Isolate version. [-Wdeprecated-declarations]
   return val->NAME ## Value(isolate->GetCurrentContext());                     \
                                                         ^
../node_modules/nan/nan_converters_43_inl.h:40:1: note: in expansion of macro ‘X’
 X(bool, Boolean)
 ^
In file included from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8-internal.h:14:0,
                 from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:25,
                 from ../src/libxmljs.cc:3:
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:2562:51: note: declared here
                 V8_WARN_UNUSED_RESULT Maybe<bool> BooleanValue(
                                                   ^
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8config.h:307:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../node_modules/nan/nan_new.h:189:0,
                 from ../node_modules/nan/nan.h:203,
                 from ../src/libxmljs.h:7,
                 from ../src/libxmljs.cc:7:
../node_modules/nan/nan_implementation_12_inl.h: In static member function ‘static Nan::imp::FactoryBase<v8::Function>::return_t Nan::imp::Factory<v8::Function>::New(Nan::FunctionCallback, v8::Local<v8::Value>)’:
../node_modules/nan/nan_implementation_12_inl.h:105:32: error: no matching function for call to ‘v8::Function::New(v8::Isolate*&, void (&)(const v8::FunctionCallbackInfo<v8::Value>&), v8::Local<v8::Object>&)’
                           , obj));
                                ^
In file included from ../src/libxmljs.cc:3:0:
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:4125:31: note: candidate: static v8::MaybeLocal<v8::Function> v8::Function::New(v8::Local<v8::Context>, v8::FunctionCallback, v8::Local<v8::Value>, int, v8::ConstructorBehavior, v8::SideEffectType)
   static MaybeLocal<Function> New(
                               ^~~
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:4125:31: note:   no known conversion for argument 1 from ‘v8::Isolate*’ to ‘v8::Local<v8::Context>’
In file included from ../node_modules/nan/nan_new.h:189:0,
                 from ../node_modules/nan/nan.h:203,
                 from ../src/libxmljs.h:7,
                 from ../src/libxmljs.cc:7:
../node_modules/nan/nan_implementation_12_inl.h: In static member function ‘static Nan::imp::FactoryBase<v8::StringObject>::return_t Nan::imp::Factory<v8::StringObject>::New(v8::Local<v8::String>)’:
../node_modules/nan/nan_implementation_12_inl.h:337:37: error: no matching function for call to ‘v8::StringObject::New(v8::Local<v8::String>&)’
   return v8::StringObject::New(value).As<v8::StringObject>();
                                     ^
In file included from ../src/libxmljs.cc:3:0:
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:5380:23: note: candidate: static v8::Local<v8::Value> v8::StringObject::New(v8::Isolate*, v8::Local<v8::String>)
   static Local<Value> New(Isolate* isolate, Local<String> value);
                       ^~~
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:5380:23: note:   candidate expects 2 arguments, 1 provided
In file included from ../node_modules/nan/nan_new.h:189:0,
                 from ../node_modules/nan/nan.h:203,
                 from ../src/libxmljs.h:7,
                 from ../src/libxmljs.cc:7:
../node_modules/nan/nan_implementation_12_inl.h:337:58: error: expected primary-expression before ‘>’ token
   return v8::StringObject::New(value).As<v8::StringObject>();
                                                          ^
../node_modules/nan/nan_implementation_12_inl.h:337:60: error: expected primary-expression before ‘)’ token
   return v8::StringObject::New(value).As<v8::StringObject>();
                                                            ^
In file included from ../src/libxmljs.h:7:0,
                 from ../src/libxmljs.cc:7:
../node_modules/nan/nan.h: In constructor ‘Nan::Utf8String::Utf8String(v8::Local<v8::Value>)’:
../node_modules/nan/nan.h:1034:53: error: no matching function for call to ‘v8::Value::ToString()’
       v8::Local<v8::String> string = from->ToString();
                                                     ^
In file included from ../src/libxmljs.cc:3:0:
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:2528:44: note: candidate: v8::MaybeLocal<v8::String> v8::Value::ToString(v8::Local<v8::Context>) const
   V8_WARN_UNUSED_RESULT MaybeLocal<String> ToString(
                                            ^~~~~~~~
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:2528:44: note:   candidate expects 1 argument, 0 provided
In file included from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8-internal.h:14:0,
                 from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:25,
                 from ../src/libxmljs.cc:3:
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:2544:35: note: candidate: v8::Local<v8::String> v8::Value::ToString(v8::Isolate*) const
                     Local<String> ToString(Isolate* isolate) const);
                                   ^
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8config.h:322:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:2544:35: note:   candidate expects 1 argument, 0 provided
                     Local<String> ToString(Isolate* isolate) const);
                                   ^
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8config.h:322:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../src/libxmljs.h:7:0,
                 from ../src/libxmljs.cc:7:
../node_modules/nan/nan.h:1044:74: error: no matching function for call to ‘v8::String::WriteUtf8(char*&, int, int, const int&)’
         length_ = string->WriteUtf8(str_, static_cast<int>(len), 0, flags);
                                                                          ^
In file included from ../src/libxmljs.cc:3:0:
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:2738:7: note: candidate: int v8::String::WriteUtf8(v8::Isolate*, char*, int, int*, int) const
   int WriteUtf8(Isolate* isolate, char* buffer, int length = -1,
       ^~~~~~~~~
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:2738:7: note:   no known conversion for argument 1 from ‘char*’ to ‘v8::Isolate*’
In file included from ../src/libxmljs.h:7:0,
                 from ../src/libxmljs.cc:7:
../node_modules/nan/nan.h: In member function ‘void Nan::AsyncWorker::SaveToPersistent(const char*, const v8::Local<v8::Value>&)’:
../node_modules/nan/nan.h:1818:64: warning: ‘bool v8::Object::Set(v8::Local<v8::Value>, v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     New(persistentHandle)->Set(New(key).ToLocalChecked(), value);
                                                                ^
In file included from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8-internal.h:14:0,
                 from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:25,
                 from ../src/libxmljs.cc:3:
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:3359:26: note: declared here
                     bool Set(Local<Value> key, Local<Value> value));
                          ^
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8config.h:322:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../src/libxmljs.h:7:0,
                 from ../src/libxmljs.cc:7:
../node_modules/nan/nan.h: In member function ‘void Nan::AsyncWorker::SaveToPersistent(const v8::Local<v8::String>&, const v8::Local<v8::Value>&)’:
../node_modules/nan/nan.h:1824:42: warning: ‘bool v8::Object::Set(v8::Local<v8::Value>, v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     New(persistentHandle)->Set(key, value);
                                          ^
In file included from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8-internal.h:14:0,
                 from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:25,
                 from ../src/libxmljs.cc:3:
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:3359:26: note: declared here
                     bool Set(Local<Value> key, Local<Value> value));
                          ^
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8config.h:322:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../src/libxmljs.h:7:0,
                 from ../src/libxmljs.cc:7:
../node_modules/nan/nan.h: In member function ‘void Nan::AsyncWorker::SaveToPersistent(uint32_t, const v8::Local<v8::Value>&)’:
../node_modules/nan/nan.h:1830:44: warning: ‘bool v8::Object::Set(uint32_t, v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     New(persistentHandle)->Set(index, value);
                                            ^
In file included from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8-internal.h:14:0,
                 from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:25,
                 from ../src/libxmljs.cc:3:
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:3368:26: note: declared here
                     bool Set(uint32_t index, Local<Value> value));
                          ^
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8config.h:322:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../src/libxmljs.h:7:0,
                 from ../src/libxmljs.cc:7:
../node_modules/nan/nan.h: In member function ‘v8::Local<v8::Value> Nan::AsyncWorker::GetFromPersistent(const char*) const’:
../node_modules/nan/nan.h:1836:61: warning: ‘v8::Local<v8::Value> v8::Object::Get(v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
         New(persistentHandle)->Get(New(key).ToLocalChecked()));
                                                             ^
In file included from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8-internal.h:14:0,
                 from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:25,
                 from ../src/libxmljs.cc:3:
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:3412:55: note: declared here
   V8_DEPRECATE_SOON("Use maybe version", Local<Value> Get(Local<Value> key));
                                                       ^
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8config.h:322:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../src/libxmljs.h:7:0,
                 from ../src/libxmljs.cc:7:
../node_modules/nan/nan.h: In member function ‘v8::Local<v8::Value> Nan::AsyncWorker::GetFromPersistent(const v8::Local<v8::String>&) const’:
../node_modules/nan/nan.h:1842:55: warning: ‘v8::Local<v8::Value> v8::Object::Get(v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     return scope.Escape(New(persistentHandle)->Get(key));
                                                       ^
In file included from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8-internal.h:14:0,
                 from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:25,
                 from ../src/libxmljs.cc:3:
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:3412:55: note: declared here
   V8_DEPRECATE_SOON("Use maybe version", Local<Value> Get(Local<Value> key));
                                                       ^
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8config.h:322:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../src/libxmljs.h:7:0,
                 from ../src/libxmljs.cc:7:
../node_modules/nan/nan.h: In member function ‘v8::Local<v8::Value> Nan::AsyncWorker::GetFromPersistent(uint32_t) const’:
../node_modules/nan/nan.h:1847:57: warning: ‘v8::Local<v8::Value> v8::Object::Get(uint32_t)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     return scope.Escape(New(persistentHandle)->Get(index));
                                                         ^
In file included from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8-internal.h:14:0,
                 from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:25,
                 from ../src/libxmljs.cc:3:
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:3416:55: note: declared here
   V8_DEPRECATE_SOON("Use maybe version", Local<Value> Get(uint32_t index));
                                                       ^
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8config.h:322:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from /usr/include/c++/7/cassert:44:0,
                 from /home/zyzzzz/.node-gyp/12.4.0/include/node/node_object_wrap.h:26,
                 from ../node_modules/nan/nan.h:53,
                 from ../src/libxmljs.h:7,
                 from ../src/libxmljs.cc:7:
../node_modules/nan/nan_object_wrap.h: In destructor ‘virtual Nan::ObjectWrap::~ObjectWrap()’:
../node_modules/nan/nan_object_wrap.h:24:25: error: ‘class Nan::Persistent<v8::Object>’ has no member named ‘IsNearDeath’
     assert(persistent().IsNearDeath());
                         ^
In file included from ../node_modules/nan/nan.h:2657:0,
                 from ../src/libxmljs.h:7,
                 from ../src/libxmljs.cc:7:
../node_modules/nan/nan_object_wrap.h: In member function ‘void Nan::ObjectWrap::MakeWeak()’:
../node_modules/nan/nan_object_wrap.h:67:34: warning: ‘void v8::PersistentBase<T>::MarkIndependent() [with T = v8::Object]’ is deprecated: Weak objects are always considered independent. Use TracedGlobal when trying to use EmbedderHeapTracer. Use a strong handle when trying to keep an object alive. [-Wdeprecated-declarations]
     persistent().MarkIndependent();
                                  ^
In file included from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8-internal.h:14:0,
                 from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:25,
                 from ../src/libxmljs.cc:3:
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:567:22: note: declared here
       V8_INLINE void MarkIndependent());
                      ^
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8config.h:307:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from /usr/include/c++/7/cassert:44:0,
                 from /home/zyzzzz/.node-gyp/12.4.0/include/node/node_object_wrap.h:26,
                 from ../node_modules/nan/nan.h:53,
                 from ../src/libxmljs.h:7,
                 from ../src/libxmljs.cc:7:
../node_modules/nan/nan_object_wrap.h: In static member function ‘static void Nan::ObjectWrap::WeakCallback(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)’:
../node_modules/nan/nan_object_wrap.h:124:26: error: ‘class Nan::Persistent<v8::Object>’ has no member named ‘IsNearDeath’
     assert(wrap->handle_.IsNearDeath());
                          ^
In file included from ../src/libxmljs.cc:8:0:
../src/xml_document.h: At global scope:
../src/xml_document.h:23:32: error: ‘v8::Handle’ has not been declared
     static void Initialize(v8::Handle<v8::Object> target);
                                ^~~~~~
../src/xml_document.h:23:38: error: expected ‘,’ or ‘...’ before ‘<’ token
     static void Initialize(v8::Handle<v8::Object> target);
                                      ^
In file included from ../src/libxmljs.cc:9:0:
../src/xml_node.h:32:32: error: ‘v8::Handle’ has not been declared
     static void Initialize(v8::Handle<v8::Object> target);
                                ^~~~~~
../src/xml_node.h:32:38: error: expected ‘,’ or ‘...’ before ‘<’ token
     static void Initialize(v8::Handle<v8::Object> target);
                                      ^
In file included from ../src/libxmljs.cc:10:0:
../src/xml_sax_parser.h:16:18: error: ‘v8::Handle’ has not been declared
   Initialize(v8::Handle<v8::Object> target);
                  ^~~~~~
../src/xml_sax_parser.h:16:24: error: expected ‘,’ or ‘...’ before ‘<’ token
   Initialize(v8::Handle<v8::Object> target);
                        ^
In file included from ../src/libxmljs.cc:11:0:
../src/xml_namespace.h:19:32: error: ‘v8::Handle’ has not been declared
     static void Initialize(v8::Handle<v8::Object> target);
                                ^~~~~~
../src/xml_namespace.h:19:38: error: expected ‘,’ or ‘...’ before ‘<’ token
     static void Initialize(v8::Handle<v8::Object> target);
                                      ^
../src/libxmljs.cc: In function ‘void libxmljs::init(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)’:
../src/libxmljs.cc:256:37: error: no matching function for call to ‘libxmljs::XmlDocument::Initialize(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE&)’
       XmlDocument::Initialize(target);
                                     ^
In file included from ../src/libxmljs.cc:8:0:
../src/xml_document.h:23:17: note: candidate: static void libxmljs::XmlDocument::Initialize(int)
     static void Initialize(v8::Handle<v8::Object> target);
                 ^~~~~~~~~~
../src/xml_document.h:23:17: note:   no known conversion for argument 1 from ‘Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE {aka v8::Local<v8::Object>}’ to ‘int’
../src/libxmljs.cc:257:38: error: no matching function for call to ‘libxmljs::XmlSaxParser::Initialize(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE&)’
       XmlSaxParser::Initialize(target);
                                      ^
In file included from ../src/libxmljs.cc:10:0:
../src/xml_sax_parser.h:16:3: note: candidate: static void libxmljs::XmlSaxParser::Initialize(int)
   Initialize(v8::Handle<v8::Object> target);
   ^~~~~~~~~~
../src/xml_sax_parser.h:16:3: note:   no known conversion for argument 1 from ‘Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE {aka v8::Local<v8::Object>}’ to ‘int’
In file included from ../src/libxmljs.h:7:0,
                 from ../src/libxmljs.cc:7:
../node_modules/nan/nan.h: In instantiation of ‘void Nan::imp::SetMethodAux(T, v8::Local<v8::String>, v8::Local<v8::FunctionTemplate>, ...) [with T = v8::Local<v8::Object>]’:
../node_modules/nan/nan.h:2353:20:   required from ‘void Nan::SetMethod(HandleType<T>, const char*, Nan::FunctionCallback) [with T = v8::Object; HandleType = v8::Local; Nan::FunctionCallback = void (*)(const Nan::FunctionCallbackInfo<v8::Value>&)]’
../src/libxmljs.cc:272:54:   required from here
../node_modules/nan/nan.h:2337:3: warning: ‘bool v8::Object::Set(v8::Local<v8::Value>, v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
   recv->Set(name, GetFunction(tpl).ToLocalChecked());
   ^~~~
In file included from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8-internal.h:14:0,
                 from /home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:25,
                 from ../src/libxmljs.cc:3:
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8.h:3359:26: note: declared here
                     bool Set(Local<Value> key, Local<Value> value));
                          ^
/home/zyzzzz/.node-gyp/12.4.0/include/node/v8config.h:322:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
xmljs.target.mk:158: recipe for target 'Release/obj.target/xmljs/src/libxmljs.o' failed
make: *** [Release/obj.target/xmljs/src/libxmljs.o] Error 1
make: Leaving directory '/home/zyzzzz/PeerTube/node_modules/libxmljs/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:200:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Linux 4.15.0-51-generic
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--loglevel=http" "--module=/home/zyzzzz/PeerTube/node_modules/libxmljs/build/Release/xmljs.node" "--module_name=xmljs" "--module_path=/home/zyzzzz/PeerTube/node_modules/libxmljs/build/Release" "--napi_version=4" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v72"
gyp ERR! cwd /home/zyzzzz/PeerTube/node_modules/libxmljs
gyp ERR! node -v v12.4.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute '/usr/bin/node /usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --loglevel=http --module=/home/zyzzzz/PeerTube/node_modules/libxmljs/build/Release/xmljs.node --module_name=xmljs --module_path=/home/zyzzzz/PeerTube/node_modules/libxmljs/build/Release --napi_version=4 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v72' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/home/zyzzzz/PeerTube/node_modules/libxmljs/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:200:13)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1021:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
node-pre-gyp ERR! System Linux 4.15.0-51-generic
node-pre-gyp ERR! command "/usr/bin/node" "/home/zyzzzz/PeerTube/node_modules/libxmljs/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build" "--loglevel" "http"
node-pre-gyp ERR! cwd /home/zyzzzz/PeerTube/node_modules/libxmljs
node-pre-gyp ERR! node -v v12.4.0
node-pre-gyp ERR! node-pre-gyp -v v0.11.0
node-pre-gyp ERR! not ok


```

It is saying 404 page so could be that dependency problem .  
```
node-pre-gyp http 404 https://github.com/libxmljs/libxmljs/releases/download/v0.19.5/node-v72-linux-x64.tar.gz
node-pre-gyp WARN Tried to download(404): https://github.com/libxmljs/libxmljs/releases/download/v0.19.5/node-v72-linux-x64.tar.gz
node-pre-gyp WARN Pre-built binaries not found for libxmljs@0.19.5 and node@12.4.0 (node-v72 ABI, glibc) (falling back to source compile with node-gyp)
node-pre-gyp http 404 status code downloading tarball https://github.com/libxmljs/libxmljs/releases/download/v0.19.5/node-v72-linux-x64.tar.gz
```

I tried both in macos and ubuntu and both are giving error.

Anyway to fix it?

**Comment by Chocobozzz** [comment]
Please try with nodejs 8

**Comment by shirshak55** [comment]
@Chocobozzz hmm used node 12 is that a reason thought it was lts version.

**Comment by shirshak55** [comment]
@Chocobozzz  tried but still same error. Anyway to fix it.

**Comment by shirshak55** [comment]
@Chocobozzz  It works on Node v8 confirmed. But it don't work on Node v12 . So will change issue title and leave it open till it get fixed :)

Thanks

**Comment by Chocobozzz** [comment]
PeerTube does not support NodeJS 12. I tried to improve the documentation: https://github.com/Chocobozzz/PeerTube/commit/4c3dbb7acd1d6be4697cfd319022b9a9fae6725b

**Comment by andrewlorien** [comment]
Same error for me in Node 8:

```
Status: Downloaded newer image for node:8
...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
error /home/node/artillery/node_modules/libxmljs: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments: 
Directory: /home/node/artillery/node_modules/libxmljs
...
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
gyp ERR! System Linux 4.9.184-linuxkit
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/node/artillery/node_modules/libxmljs
gyp ERR! node -v v8.16.1
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
```

**Comment by Chocobozzz** [comment]
@andrewlorien Try with Node 10

**Comment by shirshak55** [comment]
hmm how about supporting node v12 as it is LTS.

**Comment by Chocobozzz** [comment]
Soon I guess https://github.com/libxmljs/libxmljs/commit/5131a387147b839c2030c089c4ea11a7d6bd1cb0

**Comment by millette** [comment]
Since node v12 support was merged into v0.19.6 of libxmljs and we now [depend on v0.19.7](https://github.com/Chocobozzz/PeerTube/blob/develop/package.json#L201) and given that [node 12 is now also an LTS](https://github.com/nodejs/Release#lts_schedule), should PeerTube also support node 12?

**Comment by Chocobozzz** [comment]
@millette Yes. If it does not, please create an issue.
